app/gather/assets/apps/survey/Survey.jsx

```diff
@@ -46,26 +65,62 @@ export default class Survey extends Component {
       total: props.survey.entities_count,
       labels: props.skeleton.docs,
       allPaths: paths,
-      selectedPaths: paths
+      selectedPaths: paths,
+      isConsumerActive: false,
+      activationError: null
     }
+    this.toggleConsumer = this.toggleConsumer.bind(this)
+  }
+
+  toggleConsumer () {
+    const xform = this.props.xform && this.props.xform.results.length && this.props.xform.results[0]
+    const url = getMappingTopicsAPIPath({ id: xform ? xform.kernel_id : '' })
+
+    // get list of survey topics from kernel
+    fetchUrls([{ name: 'topics', url }])
```

Comment:
To fetch data there is the `FetchUrlsContainer` component to handle all the possible errors and responses, could you use it? 

```javascript
<FetchUrlsContainer
  urls={dashboardUrls}
  targetComponent={SurveyDashboard}
  handleResponse={handleDashboardResponse}
/>
```

Also, could you move all of this to a new component?... You know, I hate the 3000 line files :innocent: